tests/baselines/reference/parserForInStatement8.js

```diff
@@ -0,0 +1,15 @@
+//// [tests/cases/conformance/parser/ecmascript5/Statements/parserForInStatement8.ts] ////
+
+//// [parserForInStatement8.ts]
+// repro from https://github.com/microsoft/TypeScript/issues/54769
+
+for (let [x = 'a' in {}] in { '': 0 }) console.log(x)
+for (let {x = 'a' in {}} in { '': 0 }) console.log(x)
+
+
+//// [parserForInStatement8.js]
+// repro from https://github.com/microsoft/TypeScript/issues/54769
+for (var _a = (void 0)[0], x = _a === void 0 ? 'a' in {} : _a in { '': 0 })
+    console.log(x);
+for (var _b = (void 0).x, x = _b === void 0 ? 'a' in {} : _b in { '': 0 })
+    console.log(x);

```

Comment:
this isn't the correct output but since binding patterns on the left side of for-in are not supported by TS today, I think that it's fine to leave it as is for now